File: 算法训练营/03-数组链表跳表/011/双指针.go
/*
 * @lc app=leetcode.cn id=11 lang=golang
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
func maxArea(height []int) int {
	max := 0
	l, r := 0, len(height)-1
	length := 0
	for l < r {
		// 选择最短的
		if height[l] < height[r] {
			length = height[l]
		} else {
			length = height[r]
		}
		// 计算面积
		area := (r - l) * length
		
		if max < area {
			max = area
		}
		// 移动指针
		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return max
}
// @lc code=end

File: 算法训练营/03-数组链表跳表/011/双指针.py
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0
        low, high = 0, len(height)-1
        while low < high:
            area = (height[low] if height[low] < height[high] else height[high]) * (high-low)
            if area > max_area:
                max_area = area
            if height[low] < height[high]:
                low += 1
            else:
                high -= 1
        return max_area
# @lc code=end